src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/CsvFormatResponseIT.java

```diff
@@ -142,6 +156,460 @@ public void fieldOrderWithScriptFields() throws IOException {
+        String query = String.format(Locale.ROOT, "select dog_name,age from %s/dog order by age", TEST_INDEX_DOG);
+        final CSVResult csvResult = executeCsvRequest(query, false);
+
+        List<String> headers = csvResult.getHeaders();
+        Assert.assertEquals(2, headers.size());
+        Assert.assertTrue("name should be on headers", headers.contains("dog_name"));
+        Assert.assertTrue("age should be on headers", headers.contains("age"));
+
+        List<String> lines = csvResult.getLines();
+        Assert.assertEquals(2, lines.size());
+        Assert.assertTrue("rex,2".equals(lines.get(0)) || "2,rex".equals(lines.get(0)));
+        Assert.assertTrue("snoopy,4".equals(lines.get(1)) || "4,snoopy".equals(lines.get(1)));
+    }
+
+    @Test
+    public void simpleSearchResultWithNestedNotFlatNoAggs() throws Exception {
+        String query = String.format(Locale.ROOT, "select name,house from %s/gotCharacters",
+                TEST_INDEX_GAME_OF_THRONES);
+        CSVResult csvResult = executeCsvRequest(query, false);
+
+        List<String> headers = csvResult.getHeaders();
+        Assert.assertEquals(2, headers.size());
+        Assert.assertTrue("name should be on headers", headers.contains("name"));
+        Assert.assertTrue("house should be on headers", headers.contains("house"));
+
+        List<String> lines = csvResult.getLines();
+        Assert.assertEquals(7, lines.size());
+        Assert.assertTrue(lines.contains("\"{firstname=Daenerys, lastname=Targaryen, ofHerName=1}\",Targaryen") ||
+                lines.contains("\"{firstname=Daenerys, ofHerName=1, lastname=Targaryen}\",Targaryen") ||
+                lines.contains("\"{lastname=Targaryen, firstname=Daenerys, ofHerName=1}\",Targaryen") ||
+                lines.contains("\"{lastname=Targaryen, ofHerName=1, firstname=Daenerys}\",Targaryen") ||
+                lines.contains("\"{ofHerName=1, lastname=Targaryen, firstname=Daenerys}\",Targaryen") ||
+                lines.contains("\"{ofHerName=1, firstname=Daenerys, lastname=Targaryen}\",Targaryen")
+        );
+        //todo: generate all options for rest 3..
+    }
+
+
+    @Ignore("headers incorrect in case of nested fields")
+    @Test
+    public void simpleSearchResultWithNestedOneFieldNotFlatNoAggs() throws Exception {
+        String query = String.format(Locale.ROOT, "select name.firstname,house from %s/gotCharacters",
+                TEST_INDEX_GAME_OF_THRONES);
+        CSVResult csvResult = executeCsvRequest(query, false);
+
+        List<String> headers = csvResult.getHeaders();
+        Assert.assertEquals(2, headers.size());
+        Assert.assertTrue("name should be on headers", headers.contains("name"));
+        Assert.assertTrue("house should be on headers", headers.contains("house"));
+
+        List<String> lines = csvResult.getLines();
+        Assert.assertEquals(7, lines.size());
+        Assert.assertTrue(lines.contains("{firstname=Daenerys},Targaryen"));
```

Comment:
Maybe another one `containsAll` for here if agree with my previous comment.

Reply:
Probably Hamcrest matcher is more readable and can help combine both for your use.

Reply:
missed this one, will update.